File: pg-management-ui/src/app/features/tenant/models/tenant-list-dto.ts
export interface TenantListDto {
  tenantId: string;
  name: string;

  roomId?: string | null;
  roomNumber?: string | null;

  contactNumber: string;
  status: 'ACTIVE' | 'MOVED OUT' | 'NO STAY';
  checkedInAt?: string | null;
  isRentPending: boolean;
  stayType?: string | null;
  lastPaymentDate?: string | null;
  overdueSince?: string | null;
  daysOverdue?: number | null;
}

File: PgManagement-WebApi/PgManagement-WebApi/DTOs/Tenant/TenantListDto.cs
namespace PgManagement_WebApi.DTOs.Tenant
{
    public class TenantListDto
    {
        public string TenantId { get; set; }
        public string Name { get; set; }

        public string? RoomId { get; set; }
        public string? RoomNumber { get; set; }

        public string ContactNumber { get; set; }

        public string Status { get; set; }   // Active / MovedOut
        public DateTime? CheckedInAt { get; set; }
        public bool IsRentPending { get; set; }

    }
}
